src/services/ChallengeService.js

```diff
@@ -1589,6 +1589,18 @@ async function updateChallenge(currentUser, challengeId, data) {
         logger.debug(`There was an error trying to update the project: ${e.message}`);
       }
     }
+
```

Comment:
![image](https://user-images.githubusercontent.com/1445965/228499705-8ff594fd-acec-4136-bf4b-be36faca79f6.png)

We have the exact same check in another section. Let's avoid code duplicate. Our last refactor extracted out common code.

If the execution branch you introduced is different from the existing branch, the common code should move to a helper method with the different conditions at the top. For example


```
if (primaryConditionsAreSatisfied()) // chalengeStatuses in [ requirements infeasible, cancelled payment failed ]
if (conditionASatisfied() || conditionBSatisfied()) {
   projectHelper.cancelProject()
}
```

This lets us immediately understand the conditions under which a project maybe cancelled. This commit introduces redundancy

Reply:
You're right! It introduces redundancy because I missed checking in the commit where I removed the duplicate code. Please check again

Reply:
We set the value of the variable `isChallengeBeingCancelled` but we never use it. Since the section you are updating has to do with challenge cancellation, we should double check if there's any need for `isChallengeBeingCancelled` and either add back functionality that we had before that we might have missed due to the refactor, or remove `isChallengeBeingCancelled`